lib/seahorn/CexHarness.cc

```diff
@@ -104,10 +104,16 @@ Constant *exprToMemSegment(Expr segment, Expr startAddr, Expr size,
     ArrayType *placeholderT = ArrayType::get(Type::getInt8PtrTy(ctx), 0);
     return ConstantArray::get(placeholderT, LLVMValueSegment);
   }
-
+  // use MemMap to extract mem segment info
   using MemMap = expr::exprMemMap::ExprMemMap;
   const MemMap m_map(segment);
   size_t elmWidth = m_map.getContentWidth();
+  if (!m_map.isValid() || elmWidth < IntegerType::MIN_INT_BITS) {
+    LOG("cex", errs() << "memhavoc: invalid memory expression: "
```

Comment:
maybe better to use `WARN` macro? Is it an expected or unexpected behavior? 

SMT solver will not produce an array value unless the value is used. So, for example, if region is first `memhavoc` but then properly initialized, the initial value of the region will not be used and will not be computed. This is expected and desired behavior. 

One thing that has changed is that @priyasiddharth added more `memhavoc` calls to fix the fuzzer, so perhaps that is why it was missed.

TL;DR: the log should probably be `WARN` and should probably be silenced by default in `cex` log level since it is expected behavior in many cases. Maybe put it under `cex_verbose` log level.

Reply:
will change the logging lvl, I wasn't sure if this is like the other special handlings. Not sure why there wasn't any division by zero errors in line `size_t blocks = std::ceil((float)blockWidth / (float)elmWidth);`